File: scripts/fix-state-brackets.js
const fs = require('fs');
const path = require('path');

const states = ['DE', 'HI', 'ID', 'KS', 'MS', 'MT', 'ND', 'OK', 'RI', 'UT', 'VT', 'WV'];

states.forEach(state => {
  const file = path.join(__dirname, '..', 'src', 'engine', 'states', state, '2025', `compute${state}2025.ts`);

  if (!fs.existsSync(file)) {
    console.log(`❌ File not found: ${file}`);
    return;
  }

  let content = fs.readFileSync(file, 'utf-8');

  // Pattern to find: const taxBeforeCredits = calculateTaxFromBrackets(variable, STATE_RULES_2025.brackets[filingStatus]);
  const pattern = new RegExp(`const taxBeforeCredits = calculateTaxFromBrackets\\\\((\\\\w+),\\\\s*${state}_RULES_2025\\\\.brackets\\\\[filingStatus\\\\]\\\\);`);
  const match = content.match(pattern);

  if (match) {
    const taxableVar = match[1];
    const oldLine = `const taxBeforeCredits = calculateTaxFromBrackets(${taxableVar}, ${state}_RULES_2025.brackets[filingStatus]);`;
    const newLines = `const fullBrackets = convertToFullBrackets(${state}_RULES_2025.brackets[filingStatus]);
  const taxBeforeCredits = calculateTaxFromBrackets(${taxableVar}, fullBrackets);`;

    content = content.replace(oldLine, newLines);
    fs.writeFileSync(file, content, 'utf-8');
    console.log(`✅ Fixed ${state}`);
  } else {
    console.log(`⚠️  Pattern not found in ${state}, checking if already fixed...`);
    // Check if it's already using fullBrackets
    if (content.includes('convertToFullBrackets') && content.includes('fullBrackets')) {
      console.log(`   ✓ ${state} already fixed`);
    } else {
      console.log(`   ❌ ${state} needs manual review`);
    }
  }
});

console.log('\n✨ Fix complete!');

File: scripts/fix-state-brackets.py
#!/usr/bin/env python3
"""
Fix state tax computation files to properly use convertToFullBrackets
"""

import os
import re

# States that need fixing
states_to_fix = ['DE', 'HI', 'ID', 'KS', 'MS', 'MT', 'ND', 'OK', 'RI', 'UT', 'VT', 'WV']

project_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

for state in states_to_fix:
    file_path = os.path.join(project_root, 'src', 'engine', 'states', state, '2025', f'compute{state}2025.ts')

    if not os.path.exists(file_path):
        print(f"❌ File not found: {file_path}")
        continue

    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Find the pattern: calculateTaxFromBrackets(variable, STATE_RULES_2025.brackets[filingStatus])
    pattern = rf'calculateTaxFromBrackets\((\w+Taxable(?:Income)?),\s*{state}_RULES_2025\.brackets\[filingStatus\]\)'

    match = re.search(pattern, content)
    if not match:
        print(f"⚠️  Pattern not found in {state}, trying alternate pattern")
        # Try alternate variable names
        pattern = rf'calculateTaxFromBrackets\((\w+),\s*{state}_RULES_2025\.brackets\[filingStatus\]\)'
        match = re.search(pattern, content)

    if match:
        taxable_var = match.group(1)

        # Replace with correct pattern
        replacement = f'''const fullBrackets = convertToFullBrackets({state}_RULES_2025.brackets[filingStatus]);
  const taxBeforeCredits = calculateTaxFromBrackets({taxable_var}, fullBrackets)'''

        # Find and replace the original line
        old_pattern = rf'const taxBeforeCredits = calculateTaxFromBrackets\({taxable_var},\s*{state}_RULES_2025\.brackets\[filingStatus\]\);'

        new_content = re.sub(old_pattern, replacement, content)

        if new_content != content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"✅ Fixed {state}")
        else:
            print(f"⚠️  No changes made to {state} (already fixed?)")
    else:
        print(f"❌ Could not find bracket calculation in {state}")

print("\n✨ Bracket fix complete!")
